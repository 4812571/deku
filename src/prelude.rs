/*! Crate prelude

[What is a prelude?](https://doc.rust-lang.org/std/prelude/)
*/
pub use crate::{
    error::DekuError, DekuContainerRead, DekuContainerWrite, DekuRead, DekuUpdate, DekuWrite,
};
pub use bitvec::{
    order::BitOrder, order::Lsb0, order::Msb0, slice::BitSlice, vec::BitVec, view::BitView,
};
