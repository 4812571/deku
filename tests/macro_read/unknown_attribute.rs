use deku::prelude::*;

#[derive(DekuRead)]
#[deku(a)]
struct Test1 {}

#[derive(DekuRead)]
struct Test2 {
    #[deku(a = 1)]
    a: u8,
}

#[derive(DekuRead)]
#[deku(a(a))]
enum Test3 {}

#[derive(DekuRead)]
enum Test4 {
    #[deku(a = "1")]
    A(),
}

#[derive(DekuRead)]
enum Test5 {
    A(#[deku(a = "1")] u8),
}

#[derive(DekuRead)]
enum Test6 {
    A {
        #[deku(a = "1")]
        a: u8,
    },
}

fn main() {}
