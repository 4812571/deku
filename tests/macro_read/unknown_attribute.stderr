error: Unknown field: `a`
 --> $DIR/unknown_attribute.rs:4:8
  |
4 | #[deku(a)]
  |        ^

error: Unknown field: `a`
 --> $DIR/unknown_attribute.rs:9:12
  |
9 |     #[deku(a = 1)]
  |            ^^^^^

error: Unknown field: `a`
  --> $DIR/unknown_attribute.rs:14:8
   |
14 | #[deku(a(a))]
   |        ^^^^

error: Unknown field: `a`
  --> $DIR/unknown_attribute.rs:19:12
   |
19 |     #[deku(a = "1")]
   |            ^^^^^^^

error: Unknown field: `a`
  --> $DIR/unknown_attribute.rs:25:14
   |
25 |     A(#[deku(a = "1")] u8),
   |              ^^^^^^^

error: Unknown field: `a`
  --> $DIR/unknown_attribute.rs:31:16
   |
31 |         #[deku(a = "1")]
   |                ^^^^^^^
